src/Ast/Helpers.elm

```diff
@@ -48,17 +48,29 @@ type alias Alias =
     String
 
 
+type alias Line =
+    Int
+
+
+type alias Column =
+    Int
+
+
+type alias Id =
+    Int
+
+
 type alias WithMeta x =
-    { meta : { line : Int, column : Int }, e : x }
+    ( Maybe Id, Line, Column, x )
```

Comment:
Since the amount of data in Meta might change I was more thinking of a solution like
```
type alias Meta x = {x | line: Int, column: Int}
type alias WithMeta x m = (x, Meta m)
-- And this to avoid repetition in the code
type alias MExp = WithMeta Expression {}
```
Then we get nice polymorphic abstraction that is decoupled from Meta contents at the same time.
Then in elm-type-inference we could add the desired id field (which honestly doesn't have much sense in the context of elm-ast) like this

```
-- This is Inference.MExp not to be confused with Ast.MExp
type alias MExp x = WithMeta Expression {id: Int}
```

Reply:
Maybe I'll go with Located for now and later we may change it